scripts/lib/node_cache.py

```diff
@@ -45,21 +48,28 @@ def setup_node_modules(npm_args=None, stdout=None, stderr=None, copy_modules=Fal
     for cmd in cmds:
         run(cmd, stdout=stdout, stderr=stderr)
 
-def do_npm_install(target_path, npm_args, stdout=None, stderr=None, copy_modules=False,
-                   success_stamp=None):
+def do_yarn_install(target_path, yarn_args, stdout=None, stderr=None, copy_modules=False,
+                    success_stamp=None):
     # type: (str, List[str], Optional[IO], Optional[IO], Optional[bool], Optional[str]) -> None
     cmds = [
-        ["rm", "-rf", target_path],
         ['mkdir', '-p', target_path],
-        ['cp', 'package.json', target_path],
+        ['cp', 'package.json', "yarn.lock", target_path],
     ]
+    cached_node_modules = os.path.join(target_path, 'node_modules')
     if copy_modules:
         print("Cached version not found! Copying node modules.")
-        cmds.append(["cp", "-rT", "prod-static/serve/node_modules",
-                     os.path.join(target_path, "node_modules")])
+        cmds.append(["cp", "-rT", "prod-static/serve/node_modules", cached_node_modules])
     else:
         print("Cached version not found! Installing node modules.")
-        cmds.append(['npm', 'install'] + npm_args + ['--prefix', target_path])
+
+        # Install yarn if not installed
+        cmds.append(["bash", "scripts/lib/third/install-yarn.sh", "--version", "0.24.6"])
```

Comment:
@sinwar FYI, this command may add some work to what you're doing with provision.